waterbutler/providers/rushfiles/provider.py

```diff
@@ -95,29 +95,81 @@ def default_headers(self) -> dict:
-        #TODO remove if can_intra_move is always false.
-        # Check parent implementation and see if it's optimal.
-        # Implement better solution if not, remove override completely if it is.
-        raise NotImplementedError
+
+        src_metadata = await self.metadata(src_path)                 
+        now = self._get_time_for_sending()
+        request_body = json.dumps({
+            'RfVirtualFile': {
+                'ShareId': self.share['id'],
+                'ParrentId': dest_path.parent.identifier,
+                'EndOfFile': src_metadata.size if src_path.is_file else 0,
+                'Tick': 0,
+                'PublicName': dest_path.name,
+                'CreationTime': now,
+                'LastAccessTime': now,
+                'LastWriteTime': now,
+                'Attributes': 32,
+            },
+            'TransmitId': str(self._generate_uuid),
+            'ClientJournalEventType': 16,
+            'DeviceId': 'waterbutler'
+        })
+        
+        async with self.request(
+            'PUT',
+            self._build_filecache_url(str(self.share['id']), 'files', src_path.identifier),
+            data=request_body,
+            headers={'Content-Type': 'application/json'},
+            expects=(200, ),
+            throws=exceptions.IntraMoveError,
+        ) as response:
+            resp = await response.json()
+            data = resp['Data']['ClientJournalEvent']['RfVirtualFile']
+        
+        dest_path.rename(data['PublicName'])
+        
+        if dest_path.is_dir:
+            metadata = RushFilesFolderMetadata(data, dest_path)
+            metadata._children = await self._folder_metadata(dest_path)
+            return metadata, True
+        
+        data['UploadName'] = src_metadata.upload_name
+        return RushFilesFileMetadata(data, dest_path), True
 
     async def intra_copy(self,
                          dest_provider: provider.BaseProvider,
                          src_path: WaterButlerPath,
                          dest_path: WaterButlerPath) -> Tuple[RushFilesFileMetadata, bool]:
-        #TODO remove if can_intra_copy is always false
-        raise NotImplementedError
 
+        # only file
+        async with self.request(
+            'POST',
+            self._build_filecache_url(str(self.share['id']), 'files', src_path.identifier, 'clone'),
+            data=json.dumps({
+                'DestinationParentId': dest_path.identifier,
+                'DeviceId': "waterbutler",
+                'DestinationShareId': dest_provider.share['id']
+            }),
+            headers={'Content-Type': 'application/json'},
+            expects=(201, ),
+            throws=exceptions.IntraCopyError,
+        ) as response:
+            resp = await response.json()
+            data = resp['Data']['ClientJournalEvent']['RfVirtualFile']
+        
+        dest_path.rename(data['PublicName'])
```

Comment:
Rather then renaming `dest_path`, newly created path should be renamed to match `dest_path.name`

Reply:
Here however I believe my comment still stands. `.../clone` endpoint will create a new file but with the same name (+ `(duplicate...)`) as the source. However, copying in general supports also changing the name, so after creating the clone, it should be renamed to dest_path.name. Only then we can rename the destination if we get duplicate.

Reply:
OK, you're right. However, `... /clone` endpoint doesn't have `RfVirtualFile` and can't set a new name so I'll set `can_intra_copy` to false and remove this method, how about that?

Reply:
While correct, it would be much less efficient. You use `intra_move` to easily rename cloned file.

Reply:
Resolving discussion to start another about a larger piece of code instead.